### PR TITLE
KEYCLOAK-18815 Update MapKeycloakTransaction return types to match MapStorage

### PR DESCRIPTION
**Pull request by hmlnarik**

### CODE REVIEW

File: model/map/src/main/java/org/keycloak/models/map/storage/MapKeycloakTransaction.java [hunk 1]
```diff
@@ -98,17 +100,19 @@ default V getUpdated(V orig) {
...
      * @see AbstractEntity#getId()
      */
-    void updateIfChanged(V value, Predicate<V> shouldPut);
+    V updateIfChanged(V value, Predicate<V> shouldPut);
 
     /**
      * Instructs this transaction to delete a value associated with the identifier {@code key} from the underlying store
      * on commit.
      *
+     * @return Returns {@code true} if the object has been deleted or result cannot be determined, {@code false} otherwise.
```
Comment (vramik) [inline model/map/src/main/java/org/keycloak/models/map/storage/MapKeycloakTransaction.java:112]: Could you please provide an example of such situation when the result wouldn't be possible determine? Do I get it right that we might not know the result yet in moment when this method is called?

**Review by vramik — APPROVED** [review]
Thank you @hmlnarik. it looks good o me. I do have just one question, please see inline.

File: model/map/src/main/java/org/keycloak/models/map/storage/MapKeycloakTransaction.java (hunk 1 above)
Comment (hmlnarik) [reply model/map/src/main/java/org/keycloak/models/map/storage/MapKeycloakTransaction.java:112]: For example upon async deletes (when the caller does is not interested in the result due to some other eventual consistency guarantee)

**Review by mhajas — APPROVED** [review]
@hmlnarik The changes look good to me. Thanks!

**Pull request merged**